### PR TITLE
Switch to using Kokkos::{min,max} post 4.3

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

File: src/details/ArborX_DetailsBruteForceImpl.hpp [hunk 1]
```diff
@@ -22,6 +21,7 @@
 #include <ArborX_Predicates.hpp>
 
 #include <Kokkos_Core.hpp>
+#include <Kokkos_MinMax.hpp>
```
Comment (dalg24) [inline src/details/ArborX_DetailsBruteForceImpl.hpp:24]:
```suggestion
```
is provided by `<Kokkos_Core.hpp>`

File: src/details/ArborX_DetailsDistributedTreeNearest.hpp [hunk 2]
```diff
@@ -15,12 +15,12 @@
 #include <ArborX_DetailsDistributedTreeNearestHelpers.hpp>
 #include <ArborX_DetailsDistributedTreeUtils.hpp>
 #include <ArborX_DetailsKokkosExtKernelStdAlgorithms.hpp>
-#include <ArborX_DetailsKokkosExtMinMaxOperations.hpp>
 #include <ArborX_DetailsKokkosExtStdAlgorithms.hpp>
 #include <ArborX_DetailsKokkosExtViewHelpers.hpp>
 #include <ArborX_Predicates.hpp>
 
 #include <Kokkos_Core.hpp>
+#include <Kokkos_MinMax.hpp>
```
Comment (dalg24) [inline src/details/ArborX_DetailsDistributedTreeNearest.hpp:23]:
```suggestion
```

File: src/details/ArborX_DetailsDistributedTreeUtils.hpp [hunk 3]
```diff
@@ -15,13 +15,13 @@
...
 #include <ArborX_DetailsContainers.hpp>
 #include <ArborX_DetailsDistributor.hpp>
-#include <ArborX_DetailsKokkosExtMinMaxOperations.hpp>
 #include <ArborX_DetailsKokkosExtSort.hpp>
 #include <ArborX_DetailsKokkosExtStdAlgorithms.hpp>
 #include <ArborX_DetailsKokkosExtViewHelpers.hpp>
 #include <ArborX_DetailsPriorityQueue.hpp>
 
 #include <Kokkos_Core.hpp>
+#include <Kokkos_MinMax.hpp>
```
Comment (dalg24) [inline src/details/ArborX_DetailsDistributedTreeUtils.hpp:24]:
```suggestion
```

File: src/details/ArborX_DetailsMinimumSpanningTree.hpp [hunk 4]
```diff
@@ -15,14 +15,14 @@
...
 #include <ArborX_DetailsKokkosExtArithmeticTraits.hpp>
-#include <ArborX_DetailsKokkosExtMinMaxOperations.hpp>
 #include <ArborX_DetailsKokkosExtSwap.hpp>
 #include <ArborX_DetailsUtils.hpp>
 #include <ArborX_DetailsWeightedEdge.hpp>
 
 #include <Kokkos_BitManipulation.hpp>
 #include <Kokkos_Core.hpp>
 #include <Kokkos_MathematicalFunctions.hpp> // isfinite, signbit
+#include <Kokkos_MinMax.hpp>
```
Comment (dalg24) [inline src/details/ArborX_DetailsMinimumSpanningTree.hpp:25]:
```suggestion
```

File: src/details/ArborX_DetailsMortonCode.hpp [hunk 5]
```diff
@@ -12,10 +12,10 @@
 #ifndef ARBORX_DETAILS_MORTON_CODE_UTILS_HPP
 #define ARBORX_DETAILS_MORTON_CODE_UTILS_HPP
 
-#include <ArborX_DetailsKokkosExtMinMaxOperations.hpp> // min. max
 #include <ArborX_GeometryTraits.hpp>
 
 #include <Kokkos_Core.hpp> // abort
+#include <Kokkos_MinMax.hpp>
```
Comment (dalg24) [inline src/details/ArborX_DetailsMortonCode.hpp:18]:
```suggestion
#include <Kokkos_Abort.hpp>
#include <Kokkos_MinMax.hpp>
```

File: src/details/ArborX_DetailsMinimumSpanningTree.hpp (hunk 4 above)
Comment (dalg24) [reply src/details/ArborX_DetailsMinimumSpanningTree.hpp:25]: Also means this PR conflicts with #1139 
Not sure about opening both concurrently.  Would have been better to either merge or do one after the other,

File: src/details/ArborX_DetailsMinimumSpanningTree.hpp [hunk 6]
```diff
@@ -357,10 +357,9 @@ void updateLowerBounds(ExecutionSpace const &space, Labels const &labels,
   Kokkos::parallel_for(
       "ArborX::MST::update_lower_bounds",
       Kokkos::RangePolicy<ExecutionSpace>(space, 0, n), KOKKOS_LAMBDA(int i) {
-        using KokkosExt::max;
```
Comment (dalg24) [inline src/details/ArborX_DetailsMinimumSpanningTree.hpp:360]: FWIW I liked slightly better the previous version with an using-declaration.

File: src/geometry/ArborX_HyperBox.hpp [hunk 7]
```diff
@@ -13,11 +13,11 @@
 #define ARBORX_HYPERBOX_HPP
 
 #include <ArborX_DetailsKokkosExtArithmeticTraits.hpp>
-#include <ArborX_DetailsKokkosExtMinMaxOperations.hpp>
 #include <ArborX_GeometryTraits.hpp>
 #include <ArborX_HyperPoint.hpp>
 
 #include <Kokkos_Macros.hpp>
+#include <Kokkos_MinMax.hpp>
```
Comment (dalg24) [inline src/geometry/ArborX_HyperBox.hpp:20]: Where is it used?

File: src/geometry/ArborX_KDOP.hpp [hunk 8]
```diff
@@ -227,9 +228,6 @@ struct expand<KDOPTag, PointTag, KDOP, Point>
 {
   KOKKOS_FUNCTION static void apply(KDOP &kdop, Point const &point)
   {
-    using Details::KokkosExt::max;
-    using Details::KokkosExt::min;
```
Comment (dalg24) [inline src/geometry/ArborX_KDOP.hpp:231]: Please be consistent within the same file.
Either do using-declaration + unqualified calls or everywhere qualified calls.
Don't mix unless there is a good reason to.

**Review by dalg24 — CHANGES_REQUESTED** [review]
Conflicts

File: src/details/ArborX_DetailsMinimumSpanningTree.hpp (hunk 6 above)
Comment (aprokop) [reply src/details/ArborX_DetailsMinimumSpanningTree.hpp:360]: We are in control of the data it is operating on here, so I don't think it matters.

**Comment by aprokop** [comment]
> Not sure about opening both concurrently. Would have been better to either merge or do one after the other,

OK. I'll wait till merging of the swap, and will rebase this.

**Comment by aprokop** [comment]
Both CPU nodes jobs failed for unrelated reasons:
```
Cannot contact CpuNode6: hudson.remoting.ChannelClosedException:
Remote call on JNLP4-connect connection failed. The channel is
closing down or has closed down
```

**Review by dalg24 — APPROVED** [review]

**Comment by aprokop** [comment]
I'm going to merge with CPU failures. It passed CPU tests on my MAC.

**Pull request merged**